### PR TITLE
Add feature toggle repository

### PR DESCRIPTION
**Pull request by byliuyang**

### CODE REVIEW

File: backend/app/adapter/db/bool.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package db
+
+func SQLBool(value bool) int {
```
Comment (byliuyang) [inline backend/app/adapter/db/bool.go:3]:
```suggestion
// SQLBool converts Go lang bool to the corresponding SQL data type representing boolean.
func SQLBool(value bool) int {
```

File: backend/app/adapter/db/featuretoggle.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	"fmt"
+
+	"github.com/short-d/short/app/adapter/db/table"
+	"github.com/short-d/short/app/entity"
+	"github.com/short-d/short/app/usecase/repository"
+)
+
+var _ repository.FeatureToggle = (*FeatureToggleSQL)(nil)
+
+type FeatureToggleSQL struct {
```
Comment (byliuyang) [inline backend/app/adapter/db/featuretoggle.go:14]:
```suggestion
// FeatureToggleSQL accesses feature toggle information in feature_toggle table through SQL.
type FeatureToggleSQL struct {
```

File: backend/app/adapter/db/featuretoggle.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	db *sql.DB
+}
+
+func (f FeatureToggleSQL) FindToggleByID(id string) (entity.Toggle, error) {
```
Comment (byliuyang) [inline backend/app/adapter/db/featuretoggle.go:18]:
```suggestion
// FindToggleByID fetches feature toggle from the database given toggle id.
func (f FeatureToggleSQL) FindToggleByID(id string) (entity.Toggle, error) {
```

File: backend/app/adapter/db/featuretoggle.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+SELECT "%s","%s" 
+FROM "%s"
+WHERE "%s"=$1;`,
+		table.FeatureToggle.ColumnToggleID,
+		table.FeatureToggle.ColumnIsEnabled,
+		table.FeatureToggle.TableName,
+		table.FeatureToggle.ColumnToggleID,
+	)
+
+	fmt.Println(query)
```
Comment (arberiii) [inline backend/app/adapter/db/featuretoggle.go:29]: Do we need this?

File: backend/app/adapter/db/featuretoggle.go [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+	toggle := entity.Toggle{}
+	err := f.db.QueryRow(query, id).Scan(&toggle.ID, &toggle.IsEnabled)
+	return toggle, err
+}
+
+func NewFeatureToggleSQL(db *sql.DB) FeatureToggleSQL {
```
Comment (byliuyang) [inline backend/app/adapter/db/featuretoggle.go:36]:
```suggestion
// NewFeatureToggleSQL create FeatureToggleSQL repository.
func NewFeatureToggleSQL(db *sql.DB) FeatureToggleSQL {
```

File: backend/app/entity/toggle.go [hunk 6]
```diff
@@ -0,0 +1,6 @@
+package entity
+
+type Toggle struct {
```
Comment (byliuyang) [inline backend/app/entity/toggle.go:3]:
```suggestion
// Toggle represents a controllable switch that can be turned on or off.
type Toggle struct {
```

File: backend/app/usecase/repository/featuretoggle.go [hunk 7]
```diff
@@ -0,0 +1,7 @@
+package repository
+
+import "github.com/short-d/short/app/entity"
+
+type FeatureToggle interface {
```
Comment (byliuyang) [inline backend/app/usecase/repository/featuretoggle.go:5]:
```suggestion
// FeatureToggle accesses feature toggle from storage, such as database.
type FeatureToggle interface {
```

File: backend/app/adapter/db/featuretoggle.go (hunk 4 above)
Comment (byliuyang) [reply backend/app/adapter/db/featuretoggle.go:31]: Nope!

File: backend/app/adapter/db/bool.go [hunk 8]
```diff
@@ -0,0 +1,14 @@
+package db
+
+// SQLBool converts Go lang bool to the corresponding SQL data type representing boolean.
```
Comment (arberiii) [inline backend/app/adapter/db/bool.go:3]: Move this comment down to the function definition

File: backend/app/adapter/db/table/featuretoggle.go [hunk 9]
```diff
@@ -0,0 +1,11 @@
+package table
+
+var FeatureToggle = struct {
```
Comment (arberiii) [inline backend/app/adapter/db/table/featuretoggle.go:3]: Add documentation

**Pull request merged**